pkg/awsutils/awsutils.go

```diff
@@ -608,7 +602,37 @@ func (cache *EC2InstanceMetadataCache) getENIMetadata(eniMAC string) (ENIMetadat
 		deviceNum = 0
 	}
 
-	log.Debugf("Found ENI: %s, MAC %s, device %d, network card %d", eniID, eniMAC, deviceNum, networkCard)
+	log.Debugf("Found ENI: %s, MAC %s, device %d", eniID, eniMAC, deviceNum)
+	// Get IMDS fields for the interface
+	macImdsFields, err := cache.imds.GetMACImdsFields(ctx, eniMAC)
+	if err != nil {
+		awsAPIErrInc("GetMACImdsFields", err)
+		return ENIMetadata{}, err
+	}
+	isEFAOnlyInterface := true
+	// Efa-only interfaces do not have any ipv4s or ipv6s associated with it
+	for _, field := range macImdsFields {
+		if field == "local-ipv4s" {
+			imdsIPv4s, err := cache.imds.GetLocalIPv4s(ctx, eniMAC)
+			if err != nil {
+				awsAPIErrInc("GetLocalIPv4s", err)
+				return ENIMetadata{}, err
+			}
+			if len(imdsIPv4s) > 0 {
+				isEFAOnlyInterface = false
```

Comment:
My concern is if the ENI is truly missing `local-ipv4`

Reply:
> field == "local-ipv4s"

I agree. If these kind of 'structural' definitions is supported by any documentation or spec, that will be best to rely upon.